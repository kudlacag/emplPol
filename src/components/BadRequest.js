import React from "react";
// import { Link } from "react-router-dom";
import notFoundImage from "../images/404.jpg";

function BadRequest() {
  return (
    <div style={{ textAlign: "center", fontSize: "25px", padding: "50px" }}>
      <p data-testid="Text">
        There is nothing to show here or the page is loading
      </p>
      <img
        data-testid="Image"
        src={notFoundImage}
        width="550px"
        height="auto"
        alt="userpage"
      />
    </div>
  );
}

export default BadRequest;
