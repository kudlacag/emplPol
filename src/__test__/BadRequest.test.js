import BadRequest from "../components/BadRequest";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router";

describe("BadRequest", () => {
  it("checks the input values ", () => {
    let component = render(
      <MemoryRouter>
        <BadRequest />
      </MemoryRouter>
    );
    let Login = component.getByTestId("Login");
    let Text = component.getByText(
      "There is nothing to show here or the page is loading"
    );

    fireEvent.click(Login);
    expect(Login).toBeInTheDocument();
    expect(Text).toBeInTheDocument();
    expect(Login).toBeInTheDocument();
  });
});

describe("BadRequest Snapshot", () => {
  it("matches the snapshot ", () => {
    let component = render(
      <MemoryRouter>
        <BadRequest />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
