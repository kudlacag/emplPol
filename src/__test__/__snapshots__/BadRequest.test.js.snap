// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadRequest Snapshot matches the snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="text-align: center; font-size: 25px; padding: 50px;"
      >
        <p>
          There is nothing to show here or the page is loading
        </p>
        <a
          href="/login"
        >
          <button
            class="btn waves-effect waves-light"
            data-testid="Login"
          >
            Login
            <i
              class="material-icons right"
            >
              send
            </i>
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="text-align: center; font-size: 25px; padding: 50px;"
    >
      <p>
        There is nothing to show here or the page is loading
      </p>
      <a
        href="/login"
      >
        <button
          class="btn waves-effect waves-light"
          data-testid="Login"
        >
          Login
          <i
            class="material-icons right"
          >
            send
          </i>
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
